lib/add_magic_comment.rb

```diff
@@ -27,18 +28,25 @@ def self.process(options)
 
     count = 0
     EXTENSIONS.each do |ext, comment_style|
-      rbfiles = File.join(directory, "**", "*.#{ext}")
+      case ext
+        when 'erb'
+          comment = "#{MAGIC_COMMENT} %>\n"
+        else
+          comment = "#{MAGIC_COMMENT}\n"
+      end
+      frozen_literal_string_comment = comment_style.sub('{text}', comment)
+      rbfiles = File.join(directory, '**', "*.#{ext}")
       Dir.glob(rbfiles).each do |filename|
-        File.open(filename, "r+") do |file|
+        File.open(filename, 'r+') do |file|
           lines = file.readlines
 
           # remove current encoding comment(s)
-          while lines[0].match(MAGIC_COMMENT_PATTERN)
+          while lines[0]&.match(MAGIC_COMMENT_PATTERN)
```

Comment:
Good catch here for 0-length files. But I don't think we can use `&.` since that's in Ruby 2.3+. We want this tool to run even if the system Ruby is earlier than that.